libs/core/langchain_core/tools.py

```diff
@@ -586,99 +557,67 @@ async def arun(
         run_name: Optional[str] = None,
         run_id: Optional[uuid.UUID] = None,
         config: Optional[RunnableConfig] = None,
+        tool_call_id: Optional[str] = None,
         **kwargs: Any,
     ) -> Any:
         """Run the tool asynchronously."""
-        if not self.verbose and verbose is not None:
-            verbose_ = verbose
-        else:
-            verbose_ = self.verbose
         callback_manager = AsyncCallbackManager.configure(
             callbacks,
             self.callbacks,
-            verbose_,
+            self.verbose or bool(verbose),
             tags,
             self.tags,
             metadata,
             self.metadata,
         )
-        new_arg_supported = signature(self._arun).parameters.get("run_manager")
         run_manager = await callback_manager.on_tool_start(
             {"name": self.name, "description": self.description},
             tool_input if isinstance(tool_input, str) else str(tool_input),
             color=start_color,
             name=run_name,
-            inputs=tool_input,
             run_id=run_id,
+            # Inputs by definition should always be dicts.
```

Comment:
copy/pasted from elsewhere, cc @eyurtsev

Reply:
Comment is originally from observing this:

```python
@tool
def foo(x: str):
   return x

foo.invoke({'x': 'hello'})
```

It's possible that str is still supported on its own. I don't know what we want to do in that case -- since we have pretty odd behavior when it comes to langsmith API requiring auto upgrades to dicts